core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -277,10 +287,12 @@ void processSubchannelState(SubchannelData subchannelData, ConnectivityStateInfo
           if (addressIndex.increment()) {
             cancelScheduleTask();
             requestConnection(); // is recursive so might hit the end of the addresses
+          } else {
+            scheduleBackoff();
           }
         }
 
-        if (isPassComplete()) {
+        if (!firstPass || isPassComplete()) {
```

Comment:
Let's at least add a comment that the `!firstPass` is an optimization.

Reply:
Just removed it.  Looping on a boolean comparison hopefully won't take too long.